File: ADT/dfs_bfs/3577.cpp
// Tc = O(n)
// Sc = O(1)

class Solution {
public:
    int M = 1e9 + 7;

    int countPermutations(vector<int>& complexity) {
        int n = complexity.size();

        long long result = 1;

        for (int i = 1; i < n; i++) {
            if (complexity[i] <= complexity[0])
                return 0;

            result = (result * i) % M;
        }

        return result;
    }
};

File: ADT/dfs_bfs/3577.js

var countPermutations = function (complexity) {
    const M = 1e9 + 7
    const n = complexity.length
    let result = 1
    for (let i = 1; i < n; i++) {
        if (complexity[i] <= complexity[0]) {
            return 0
        }
        result = (result * i) % M
    }
    return result
};